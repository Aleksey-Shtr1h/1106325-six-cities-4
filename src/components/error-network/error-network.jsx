import React from 'react';
import {propsTypeAll} from '../../propsType/propsType.js';

const StatusError = {
  '400': {
    status: `400`,
    infoError: `Введите корректные данные`,
  },

  '401': {
    status: `401`,
    infoError: `Требуется авторизация`,
  },

  '404': {
    status: `404`,
    infoError: `Требуется авторизация`,
  },

  'any': {
    status: ``,
    infoError: `Произошла ошибка`,
  },

};

export const ErrorNetwork = ({err, onHideErrorBlock}) => {

  const error = err in StatusError ? StatusError[err] : StatusError.any;

  return (
    <section className="error-network">

      <p className="error-network__text">
        Произошла ошибка
      </p>

      <p className="error-network__status">
        {error.status}
      </p>

      <p className="error-network__infoError">
        {error.infoError}
      </p>

      <button
        type="button"
        onClick={() => onHideErrorBlock()}
      >
        Скрыть
      </button>
    </section>
  );
};

ErrorNetwork.propTypes = {
  err: propsTypeAll.numberAndNullAndString,
  onHideErrorBlock: propsTypeAll.func,
};
