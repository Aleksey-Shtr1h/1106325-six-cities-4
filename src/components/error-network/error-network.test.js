import React from 'react';
import renderer from 'react-test-renderer';

import {ErrorNetwork} from './error-network.jsx';

// import {testProps} from '../../test/offersProps.js';


it(`Should ErrorNetwork render correctly 1`, () => {

  const tree = renderer.create(
      <ErrorNetwork
        err={`test`}
        onHideErrorBlock={() => {}}
      />,
      {
        createNodeMock: () => {
          return document.createElement(`div`);
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
