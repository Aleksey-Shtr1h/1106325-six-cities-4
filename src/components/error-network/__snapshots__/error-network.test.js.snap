// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ErrorNetwork render correctly 1 1`] = `
<section
  className="error-network"
>
  <p
    className="error-network__text"
  >
    Произошла ошибка
  </p>
  <p
    className="error-network__status"
  >
    
  </p>
  <p
    className="error-network__infoError"
  >
    Произошла ошибка
  </p>
  <button
    className="error-network__btn"
    onClick={[Function]}
    type="button"
  >
    Скрыть
  </button>
</section>
`;
